ts/DataGrid/DataGrid2/DataGridTable.ts

```diff
@@ -0,0 +1,367 @@
+            this.columns.push(
+                new DataGridColumn(this, columnId, i)
+            );
+        }
+    }
+
+    /**
+     * Reflows the table's content dimensions.
+     */
+    public reflow(): void {
+        this.tbodyElement.style.height = this.tbodyElement.style.minHeight = `${
+            (this.dataGrid.container?.clientHeight || 0) -
+            this.theadElement.offsetHeight -
+            (this.titleElement?.offsetHeight || 0)
+        }px`;
+
+        // Get the width of the rows.
+        if (this.columnDistribution === 'fixed') {
+            let rowsWidth = 0;
+            for (let i = 0, iEnd = this.columns.length; i < iEnd; ++i) {
+                rowsWidth += this.columns[i].width;
+            }
+            this.rowsWidth = rowsWidth;
+        }
+
+        // Reflow the head
+        if (this.head) {
+            this.head.reflow();
+        }
+
+        // Reflow rows content dimensions
+        this.rowsVirtualizer.reflowRows();
+    }
+
+    /**
+     * Handles the resize event.
+     */
+    private onResize = (): void => {
+        this.reflow();
+    };
+
+    /**
+     * Handles the scroll event.
+     */
+    private onScroll = (): void => {
+        this.rowsVirtualizer.scroll();
+        this.head?.scrollHorizontally(this.tbodyElement.scrollLeft);
+    };
+
+    /**
+     * Scrolls the table to the specified row.
+     *
+     * @param index
+     * The index of the row to scroll to.
+     */
+    public scrollToRow(index: number): void {
+        this.tbodyElement.scrollTop =
+            index * this.rowsVirtualizer.defaultRowHeight;
+    }
+
+    /**
+     * Get the widthRatio value from the width in pixels. The widthRatio is
+     * calculated based on the width of the viewport.
+     *
+     * @param width
+     * The width in pixels.
+     *
+     * @return The width ratio.
+     */
+    public getRatioFromWidth(width: number): number {
+        return width / this.tbodyElement.clientWidth;
+    }
+
+    /**
+     * Get the width in pixels from the widthRatio value. The width is
+     * calculated based on the width of the viewport.
+     *
+     * @param ratio
+     * The width ratio.
+     *
+     * @returns The width in pixels.
+     */
+    public getWidthFromRatio(ratio: number): number {
+        return this.tbodyElement.clientWidth * ratio;
+    }
+
+    /**
+     * Renders the no result row.
+     */
+    private renderNoResultRow(): void {
+        const row = makeHTMLElement('tr', {
+            className: Globals.classNames.rowElement
+        }, this.tbodyElement);
+
+        const cell = makeHTMLElement('td', {
+            className: Globals.classNames.noResultColumn
+        }, row);
+
+        cell.innerHTML = 'No data';
+    }
+
+    /**
+     * Render title above the datagrid
+     */
+    public renderTitle(): void {
+        if (!this.dataGrid.options?.title) {
+            return;
+        }
+
+        this.titleElement = makeHTMLElement('caption', {
```

Comment:
Why this is rendered as a caption?

Reply:
Caption is the correct accessible way to add "titles" to tables (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/caption). If we want proper title (H-tag) as well I think that should be in the component scope. If used as a standalone DataGrid outside of Dashboards they can add it themselves if needed in their markup.

Buuuuuut, I think `options.caption` is more correct, but perhaps not that intuitive for some devs? Regardless, I think we should change to `options.caption` and `renderCaption`.

Reply:
The "caption" is used because its build-in table solution for it (like Stian said).
We didn't name title as caption (in options), because means - text below object (like image etc) for the end user, so when we place text above the table its title. 

It's tricky.

Reply:
But it will be a part of DataGrid API, not DataGridComponent, so I think the idea of changing it to `caption` is reasonable. I'll do it.

Reply:
> But it will be a part of DataGrid API, not DataGridComponent, so I think the idea of changing it to `caption` is reasonable. I'll do it.

I think we should add caption-side: top; and text-align: left; to the default CSS (and document it) to make it clear for users how to position it. 

https://developer.mozilla.org/en-US/docs/Web/CSS/caption-side is not a well known attribute, according to the  Internet.